src/api/app/models/project.rb

```diff
@@ -963,12 +971,34 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
 
     return unless pkg_to_enable.is_channel?
 
-    # explizit call for a channel package, so create the repos for it
+    # explicit call for a channel package, so create the repos for it
     pkg_to_enable.channels.each do |channel|
       channel.add_channel_repos_to_project(pkg_to_enable)
     end
   end
 
+  def branch_remote_repositories(project_name, pkg_to_enable)
+    # handle remote package case
+    possible_ancestors = RemoteProject.new(name: project_name).possible_ancestor_names
+    possible_ancestors = [project_name] if possible_ancestors.size == 0
+
+    interconnect_project = RemoteProject.find_by(name: possible_ancestors)
+
+    return false unless interconnect_project
+
+    # retrieve and parse remote _meta file for project
+    remote_project_meta = Nokogiri::XML(ProjectMetaFile.new(project_name: project_name).to_s)
+
+    # append the repositories to the _meta file of the local project
+    local_project_meta = Nokogiri::XML(pkg_to_enable.project.to_axml)
+    local_project_meta.at('project').add_child(remote_project_meta.xpath('//repository'))
+
+    # update branched project _meta file
+    pkg_to_enable.project.update_from_xml(Xmlhash.parse(local_project_meta.to_xml))
```

Comment:
What about the result from ``update_from_xml``? It does return a hash with an error message if it fails!